src/main/java/edu/kit/provideq/toolbox/meta/Problem.java

```diff
@@ -208,4 +233,8 @@ public String toString() {
             + ", solver=" + solver
             + '}';
   }
+
+  public Bound getBound() {
```

Comment:
Could you make this return an optional please (the bound attribute itself can stay as it is but you can use the `Optional.ofNullable()` to return the bound optional here).
I will apply the same refactoring for `getSolution`, this was an oversight before.